src/Trixi.jl

```diff
@@ -20,30 +20,46 @@ using Printf: @printf, @sprintf, println
 using Profile: clear_malloc_data
 using Random: seed!
 
-using EllipsisNotation
+using DiffEqBase: ODEProblem, ODESolution, get_du, u_modified!, set_proposed_dt!
+using DiffEqCallbacks: CallbackSet, DiscreteCallback
+using EllipsisNotation # ..
 using HDF5: h5open, attrs
 using StaticArrays: @MVector, @SVector, MVector, MMatrix, MArray, SVector, SMatrix, SArray
-using TimerOutputs: @notimeit, @timeit, TimerOutput, print_timer, reset_timer!
+using TimerOutputs: @notimeit, @timeit, @timeit_debug, TimerOutput, print_timer, reset_timer!
 using UnPack: @unpack
 
 # Tullio.jl makes use of LoopVectorization.jl via Requires.jl.
-# Hence, we need `using LoopVectorization` after loading Tullio and before using @tullio.
+# Hence, we need `using LoopVectorization` after loading Tullio and before using `@tullio`.
 using Tullio: @tullio
 using LoopVectorization
 
+
+# TODO: Taal remove globals
 # Use a central dictionary for global settings
 const globals = Dict{Symbol, Any}()
 export globals
 
+
+# TODO: Taal, decide where to define the entry points of our type hierarchy,
+# e.g. AbstractEquations, AbstractSemidiscretization etc.
+# Placing them here will allow us to make use of them for dispatch even for
+# other stuff defined very early in our include pipeline, e.g.
+#   IndicatorLöhner(semi::AbstractSemidiscretization)
```

Comment:
What are the downsides to doing this? I don't see any, except that I'd prefer to put those entry points into a separate file if it becomes more than ten lines or so, just to keep `Trixi.jl` easy to read for someone new.

Reply:
It's okay for me to put these into a separate file. The only downside could be that someone would expect to find the definition of `AbstractEquations` in `equations.jl` instead of `abstract_types.jl`.